tests/unit/modules/test_kubernetes.py

```diff
@@ -141,23 +140,49 @@ def test_create_deployments(self):
                     kubernetes.kubernetes.client.ExtensionsV1beta1Api().
                     create_namespaced_deployment().to_dict.called)
 
-    def test_setup_client_key_file(self):
+    def test_setup_kubeconfig_file(self):
         '''
-        Test that the `kubernetes.client-key-file` configuration isn't overwritten
+        Test that the `kubernetes.kubeconfig` configuration isn't overwritten
         :return:
         '''
         def settings(name, value=None):
             data = {
-                'kubernetes.client-key-file': '/home/testuser/.minikube/client.key',
+                'kubernetes.kubeconfig': '/home/testuser/.minikube/kubeconfig.cfg',
+                'kubernetes.context': 'minikube'
             }
             return data.get(name, value)
 
-        with patch.dict(kubernetes.__salt__, {'config.option': Mock(side_effect=settings)}):
-            config = kubernetes._setup_conn()
-            self.assertEqual(
-                settings('kubernetes.client-key-file'),
-                config['key_file'],
-            )
+        with mock_kubernetes_library() as mock_kubernetes_lib:
+            with patch.dict(kubernetes.__salt__, {'config.option': Mock(side_effect=settings)}):
+                mock_kubernetes_lib.config.load_kube_config = Mock()
+                config = kubernetes._setup_conn()
+                self.assertEqual(
+                    settings('kubernetes.kubeconfig'),
+                    config['kubeconfig'],
+                )
+
+    def test_setup_kubeconfig_data_overwrite(self):
+        '''
+        Test that provided `kubernetes.kubeconfig` configuration is overwritten
+        by provided kubeconfig_data in the command
+        :return:
+        '''
+        def settings(name, value=None):
+            data = {
+                'kubernetes.kubeconfig': '/home/testuser/.minikube/kubeconfig.cfg',
+                'kubernetes.context': 'minikube'
+            }
+            return data.get(name, value)
```

Comment:
Why copypaste the same thing all over again? It can be just a static method and you could do something like:

```python
    @staticmethod
    def kube_settings(name, value=None):
        data = {....}
        return data.get(name, value)

    def test......:
        with mock_kubernetes_library() ....:
            with patch.dict(kubernetes.__salt__, {'config.option': Mock(side_effect=KubernetesTestCase.kube_settings)}):
                ...
```
...and then just reuse it twice. I am not fan with multiple nested `with` statements (which you still can have multiple conditions in just _one statement_) and I would use `@patch` decorator instead, but for now it is OK.

Please refactor this.